File: scripts/generate_icons.py
import os
import subprocess

# SVG template for HOP MAP logo in 512x512 square
square_svg = """<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" width="512" height="512">
  <rect width="512" height="512" fill="#0f1422" />
  
  <g transform="translate(16, 16) scale(0.9375)">
    <!-- ==================== TOP ROW: HOP ==================== -->
    
    <!-- H letter -->
    <g id="letter-H">
      <!-- Orange background 3D extrusion / shadow -->
      <!-- Left pillar shadow -->
      <rect x="15" y="60" width="80" height="175" rx="4" fill="#ff7a00" />
      <rect x="25" y="50" width="70" height="185" rx="4" fill="#ff7a00" />
      <!-- Right pillar shadow -->
      <rect x="115" y="60" width="80" height="175" rx="4" fill="#ff7a00" />
      <rect x="105" y="70" width="90" height="165" rx="4" fill="#ff7a00" />
      <!-- Crossbar shadow -->
      <rect x="40" y="115" width="130" height="65" fill="#ff7a00" />

      <!-- Main Yellow H -->
      <path d="M 38 75 
               L 92 75 
               L 92 125 
               L 122 125 
               L 122 75 
               L 176 75 
               L 176 225 
               L 122 225 
               L 122 170 
               L 92 170 
               L 92 225 
               L 38 225 Z" 
            fill="#ffd000" stroke="#ff7a00" stroke-width="8" stroke-linejoin="miter" />
      
      <!-- Yellow inner highlights -->
      <path d="M 45 83 L 53 83 L 53 217 L 45 217 Z" fill="#fff066" opacity="0.6" />
      <path d="M 161 83 L 169 83 L 169 217 L 161 217 Z" fill="#fff066" opacity="0.6" />
      <path d="M 53 210 L 85 210 L 85 217 L 53 217 Z" fill="#fff066" opacity="0.6" />
      <path d="M 129 210 L 161 210 L 161 217 L 129 217 Z" fill="#fff066" opacity="0.6" />
    </g>

    <!-- O letter (Vinyl / Hop Disc) -->
    <g id="letter-O">
      <!-- Orange 3D extrusion offset -->
      <ellipse cx="265" cy="130" rx="80" ry="78" fill="#ff7a00" />
      <circle cx="265" cy="155" r="76" fill="#ff7a00" />
      
      <!-- Main Yellow Disc -->
      <circle cx="278" cy="148" r="74" fill="#ffd000" stroke="#ff7a00" stroke-width="8" />
      
      <!-- Center dot / hole -->
      <circle cx="278" cy="148" r="9" fill="#ff7a00" />
      
      <!-- Stylized curved groove / highlight -->
      <path d="M 330 148 A 52 52 0 0 1 298 196" fill="none" stroke="#fff066" stroke-width="7" stroke-linecap="round" />
      <path d="M 235 125 A 50 50 0 0 1 278 98" fill="none" stroke="#fff066" stroke-width="6" stroke-linecap="round" opacity="0.8" />
    </g>

    <!-- P letter (Top) -->
    <g id="letter-P-top">
      <!-- Orange shadow -->
      <path d="M 355 60 L 465 60 A 65 65 0 0 1 465 175 L 435 175 L 435 235 L 355 235 Z" fill="#ff7a00" />
      <path d="M 365 52 L 470 52 A 65 65 0 0 1 470 170 L 440 170 L 440 235 L 365 235 Z" fill="#ff7a00" />

      <!-- Main Yellow P -->
      <path d="M 378 75 
               L 440 75 
               A 42 42 0 0 1 440 160 
               L 420 160 
               L 420 225 
               L 378 225 Z" 
            fill="#ffd000" stroke="#ff7a00" stroke-width="8" stroke-linejoin="round" />
      
      <!-- P Counter (Hole / Dot) -->
      <circle cx="438" cy="118" r="9" fill="#ff7a00" />
      
      <!-- Stylized inner line -->
      <path d="M 390 85 L 420 85 A 25 25 0 0 1 420 135 L 390 135 Z" fill="none" stroke="#ff7a00" stroke-width="5" stroke-linejoin="round" />
      <path d="M 384 83 L 384 217" stroke="#fff066" stroke-width="6" stroke-linecap="round" opacity="0.6" />
    </g>


    <!-- ==================== BOTTOM ROW: MAP ==================== -->

    <!-- M letter -->
    <g id="letter-M">
      <!-- Orange 3D shadow with jagged crowns -->
      <path d="M 15 245 L 75 320 L 165 245 L 185 260 L 185 435 L 15 435 Z" fill="#ff7a00" />
      <polygon points="15,245 40,245 40,430 15,430" fill="#ff7a00" />
      <polygon points="155,245 185,245 185,430 155,430" fill="#ff7a00" />

      <!-- Main Yellow M with jagged peaks -->
      <path d="M 38 285 
               L 95 365 
               L 170 268 
               L 170 425 
               L 125 425 
               L 125 348 
               L 95 388 
               L 65 348 
               L 65 425 
               L 38 425 Z" 
            fill="#ffd000" stroke="#ff7a00" stroke-width="8" stroke-linejoin="miter" />
      
      <!-- Jagged Top-left fin -->
      <polygon points="15,245 38,285 38,415 15,350" fill="#ff7a00" />
      <polygon points="140,250 170,268 170,350 140,290" fill="#ff7a00" opacity="0.7" />

      <!-- Highlights -->
      <path d="M 45 300 L 45 418" stroke="#fff066" stroke-width="6" stroke-linecap="round" opacity="0.6" />
      <path d="M 155 285 L 155 418" stroke="#fff066" stroke-width="6" stroke-linecap="round" opacity="0.6" />
    </g>

    <!-- A letter (Triangle Slice) -->
    <g id="letter-A">
      <!-- Orange 3D shadow -->
      <polygon points="268,248 185,410 355,410" fill="#ff7a00" />
      <polygon points="280,265 195,430 365,430" fill="#ff7a00" />

      <!-- Main Yellow Triangle -->
      <polygon points="285,285 220,425 350,425" fill="#ffd000" stroke="#ff7a00" stroke-width="8" stroke-linejoin="round" />
      
      <!-- Center dot / pip -->
      <circle cx="285" cy="370" r="9" fill="#ff7a00" />
      
      <!-- Inner angular accent -->
      <polyline points="265,415 330,415 315,385" fill="none" stroke="#fff066" stroke-width="6" stroke-linecap="round" opacity="0.7" />
    </g>

    <!-- P letter (Bottom) -->
    <g id="letter-P-bottom">
      <!-- Orange shadow -->
      <path d="M 355 260 L 465 260 A 65 65 0 0 1 465 375 L 435 375 L 435 435 L 355 435 Z" fill="#ff7a00" />
      <path d="M 365 252 L 470 252 A 65 65 0 0 1 470 370 L 440 370 L 440 435 L 365 435 Z" fill="#ff7a00" />

      <!-- Main Yellow P -->
      <path d="M 378 275 
               L 440 275 
               A 42 42 0 0 1 440 360 
               L 420 360 
               L 420 425 
               L 378 425 Z" 
            fill="#ffd000" stroke="#ff7a00" stroke-width="8" stroke-linejoin="round" />
      
      <!-- P Counter (Dot / Hole) -->
      <circle cx="438" cy="318" r="9" fill="#ff7a00" />
      
      <!-- Stylized inner line -->
      <path d="M 390 285 L 420 285 A 25 25 0 0 1 420 335 L 390 335 Z" fill="none" stroke="#ff7a00" stroke-width="5" stroke-linejoin="round" />
      <path d="M 384 283 L 384 417" stroke="#fff066" stroke-width="6" stroke-linecap="round" opacity="0.6" />
    </g>
  </g>
</svg>
"""

# OG image template (1200x630)
og_svg = """<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1200 630" width="1200" height="630">
  <rect width="1200" height="630" fill="#0f1422" />
  
  <g transform="translate(344, 59)">
    <!-- Main Square 512x512 centered -->
    <g transform="translate(0, 0)">
      <!-- H letter -->
      <g id="og-letter-H">
        <rect x="15" y="60" width="80" height="175" rx="4" fill="#ff7a00" />
        <rect x="25" y="50" width="70" height="185" rx="4" fill="#ff7a00" />
        <rect x="115" y="60" width="80" height="175" rx="4" fill="#ff7a00" />
        <rect x="105" y="70" width="90" height="165" rx="4" fill="#ff7a00" />
        <rect x="40" y="115" width="130" height="65" fill="#ff7a00" />

        <path d="M 38 75 L 92 75 L 92 125 L 122 125 L 122 75 L 176 75 L 176 225 L 122 225 L 122 170 L 92 170 L 92 225 L 38 225 Z" 
              fill="#ffd000" stroke="#ff7a00" stroke-width="8" stroke-linejoin="miter" />
        
        <path d="M 45 83 L 53 83 L 53 217 L 45 217 Z" fill="#fff066" opacity="0.6" />
        <path d="M 161 83 L 169 83 L 169 217 L 161 217 Z" fill="#fff066" opacity="0.6" />
        <path d="M 53 210 L 85 210 L 85 217 L 53 217 Z" fill="#fff066" opacity="0.6" />
        <path d="M 129 210 L 161 210 L 161 217 L 129 217 Z" fill="#fff066" opacity="0.6" />
      </g>

      <!-- O letter -->
      <g id="og-letter-O">
        <ellipse cx="265" cy="130" rx="80" ry="78" fill="#ff7a00" />
        <circle cx="265" cy="155" r="76" fill="#ff7a00" />
        <circle cx="278" cy="148" r="74" fill="#ffd000" stroke="#ff7a00" stroke-width="8" />
        <circle cx="278" cy="148" r="9" fill="#ff7a00" />
        <path d="M 330 148 A 52 52 0 0 1 298 196" fill="none" stroke="#fff066" stroke-width="7" stroke-linecap="round" />
        <path d="M 235 125 A 50 50 0 0 1 278 98" fill="none" stroke="#fff066" stroke-width="6" stroke-linecap="round" opacity="0.8" />
      </g>

      <!-- P letter top -->
      <g id="og-letter-P-top">
        <path d="M 355 60 L 465 60 A 65 65 0 0 1 465 175 L 435 175 L 435 235 L 355 235 Z" fill="#ff7a00" />
        <path d="M 365 52 L 470 52 A 65 65 0 0 1 470 170 L 440 170 L 440 235 L 365 235 Z" fill="#ff7a00" />
        <path d="M 378 75 L 440 75 A 42 42 0 0 1 440 160 L 420 160 L 420 225 L 378 225 Z" 
              fill="#ffd000" stroke="#ff7a00" stroke-width="8" stroke-linejoin="round" />
        <circle cx="438" cy="118" r="9" fill="#ff7a00" />
        <path d="M 390 85 L 420 85 A 25 25 0 0 1 420 135 L 390 135 Z" fill="none" stroke="#ff7a00" stroke-width="5" stroke-linejoin="round" />
        <path d="M 384 83 L 384 217" stroke="#fff066" stroke-width="6" stroke-linecap="round" opacity="0.6" />
      </g>

      <!-- M letter -->
      <g id="og-letter-M">
        <path d="M 15 245 L 75 320 L 165 245 L 185 260 L 185 435 L 15 435 Z" fill="#ff7a00" />
        <polygon points="15,245 40,245 40,430 15,430" fill="#ff7a00" />
        <polygon points="155,245 185,245 185,430 155,430" fill="#ff7a00" />
        <path d="M 38 285 L 95 365 L 170 268 L 170 425 L 125 425 L 125 348 L 95 388 L 65 348 L 65 425 L 38 425 Z" 
              fill="#ffd000" stroke="#ff7a00" stroke-width="8" stroke-linejoin="miter" />
        <polygon points="15,245 38,285 38,415 15,350" fill="#ff7a00" />
        <polygon points="140,250 170,268 170,350 140,290" fill="#ff7a00" opacity="0.7" />
        <path d="M 45 300 L 45 418" stroke="#fff066" stroke-width="6" stroke-linecap="round" opacity="0.6" />
        <path d="M 155 285 L 155 418" stroke="#fff066" stroke-width="6" stroke-linecap="round" opacity="0.6" />
      </g>

      <!-- A letter -->
      <g id="og-letter-A">
        <polygon points="268,248 185,410 355,410" fill="#ff7a00" />
        <polygon points="280,265 195,430 365,430" fill="#ff7a00" />
        <polygon points="285,285 220,425 350,425" fill="#ffd000" stroke="#ff7a00" stroke-width="8" stroke-linejoin="round" />
        <circle cx="285" cy="370" r="9" fill="#ff7a00" />
        <polyline points="265,415 330,415 315,385" fill="none" stroke="#fff066" stroke-width="6" stroke-linecap="round" opacity="0.7" />
      </g>

      <!-- P letter bottom -->
      <g id="og-letter-P-bottom">
        <path d="M 355 260 L 465 260 A 65 65 0 0 1 465 375 L 435 375 L 435 435 L 355 435 Z" fill="#ff7a00" />
        <path d="M 365 252 L 470 252 A 65 65 0 0 1 470 370 L 440 370 L 440 435 L 365 435 Z" fill="#ff7a00" />
        <path d="M 378 275 L 440 275 A 42 42 0 0 1 440 360 L 420 360 L 420 425 L 378 425 Z" 
              fill="#ffd000" stroke="#ff7a00" stroke-width="8" stroke-linejoin="round" />
        <circle cx="438" cy="318" r="9" fill="#ff7a00" />
        <path d="M 390 285 L 420 285 A 25 25 0 0 1 420 335 L 390 335 Z" fill="none" stroke="#ff7a00" stroke-width="5" stroke-linejoin="round" />
        <path d="M 384 283 L 384 417" stroke="#fff066" stroke-width="6" stroke-linecap="round" opacity="0.6" />
      </g>
    </g>
  </g>
</svg>
"""

with open("/tmp/square.svg", "w") as f:
    f.write(square_svg)

with open("/tmp/og.svg", "w") as f:
    f.write(og_svg)

# Targets:
# Sizes for square:
# 16x16: favicon-16x16.png
# 32x32: favicon-32x32.png
# 180x180: apple-touch-icon.png
# 192x192: android-chrome-192x192.png, icon-192.png
# 512x512: android-chrome-512x512.png, icon-512.png, favicon.png
# favicon.ico (32x32 or multi-layer ico)
# og-image.png (1200x630)

files_to_generate = [
    ("/tmp/square.svg", "favicon-16x16.png", 16, 16),
    ("/tmp/square.svg", "favicon-32x32.png", 32, 32),
    ("/tmp/square.svg", "apple-touch-icon.png", 180, 180),
    ("/tmp/square.svg", "android-chrome-192x192.png", 192, 192),
    ("/tmp/square.svg", "icon-192.png", 192, 192),
    ("/tmp/square.svg", "android-chrome-512x512.png", 512, 512),
    ("/tmp/square.svg", "icon-512.png", 512, 512),
    ("/tmp/square.svg", "favicon.png", 512, 512),
    ("/tmp/square.svg", "favicon.ico", 32, 32),
    ("/tmp/og.svg", "og-image.png", 1200, 630),
]

os.makedirs("/public", exist_ok=True)

for src_svg, filename, w, h in files_to_generate:
    # Generate into /public and /
    dest_pub = os.path.join("/public", filename)
    dest_root = os.path.join("/", filename)
    
    cmd_pub = ["convert", "-background", "none", "-resize", f"{w}x{h}", src_svg, dest_pub]
    subprocess.run(cmd_pub, check=True)
    
    cmd_root = ["convert", "-background", "none", "-resize", f"{w}x{h}", src_svg, dest_root]
    subprocess.run(cmd_root, check=True)
    print(f"Generated {filename} ({w}x{h}) in /public/ and /")

print("All icons successfully generated!")

File: scripts/generate_icons.js
import sharp from 'sharp';
import fs from 'fs';
import path from 'path';

// Clean SVG without double-hyphen in comments
const svgLogo = `
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" width="512" height="512">
  <rect width="512" height="512" fill="#141724" rx="0" />

  <g id="hop-map-art">
    <!-- TOP ROW: HOP -->

    <!-- LETTER H 3D SHADOW -->
    <path d="M 64 96 L 132 96 L 132 140 L 156 140 L 156 96 L 194 96 L 194 218 L 156 218 L 156 174 L 132 174 L 132 218 L 64 218 Z" 
          fill="#FF5E00" />
    <!-- LETTER H FRONT FACE -->
    <path d="M 88 120 L 134 120 L 134 152 L 158 152 L 158 120 L 194 120 L 194 232 L 158 232 L 158 186 L 134 186 L 134 232 L 88 232 Z" 
          fill="#FFB800" stroke="#FF7700" stroke-width="4" stroke-linejoin="miter" />
    <line x1="96" y1="128" x2="96" y2="224" stroke="#FFE680" stroke-width="3" stroke-linecap="round" opacity="0.8" />
    <line x1="184" y1="128" x2="184" y2="224" stroke="#FFE680" stroke-width="3" stroke-linecap="round" opacity="0.8" />
    <line x1="130" y1="168" x2="162" y2="168" stroke="#FFE680" stroke-width="2.5" stroke-linecap="round" opacity="0.8" />
    <path d="M 166 128 L 186 128 L 186 224 L 166 224" fill="none" stroke="#FF7700" stroke-width="2" opacity="0.6" />

    <!-- LETTER O 3D SHADOW -->
    <ellipse cx="250" cy="168" rx="68" ry="68" fill="#FF5E00" />
    
    <!-- LETTER O FRONT FACE -->
    <circle cx="270" cy="174" r="62" fill="#FFB800" stroke="#FF7700" stroke-width="4" />
    <circle cx="270" cy="174" r="48" fill="none" stroke="#FF5E00" stroke-width="3.5" />
    <circle cx="270" cy="174" r="46" fill="none" stroke="#FFE680" stroke-width="2" opacity="0.7" />
    <circle cx="270" cy="174" r="8.5" fill="#FF5E00" />
    <circle cx="270" cy="174" r="5" fill="#FFE680" />
    <path d="M 230 145 A 50 50 0 0 1 310 145" fill="none" stroke="#FFE680" stroke-width="3.5" stroke-linecap="round" />

    <!-- LETTER P TOP 3D SHADOW -->
    <path d="M 338 96 L 418 96 C 444 96 458 114 458 140 C 458 166 444 182 418 182 L 382 182 L 382 218 L 338 218 Z" 
          fill="#FF5E00" />
    <!-- LETTER P TOP FRONT FACE -->
    <path d="M 356 120 L 416 120 C 438 120 452 134 452 154 C 452 174 438 188 416 188 L 392 188 L 392 232 L 356 232 Z" 
          fill="#FFB800" stroke="#FF7700" stroke-width="4" stroke-linejoin="miter" />
    <ellipse cx="410" cy="154" rx="18" ry="18" fill="#141724" stroke="#FF7700" stroke-width="3.5" />
    <circle cx="410" cy="154" r="7" fill="#FF5E00" />
    <circle cx="410" cy="154" r="4" fill="#FFE680" />
    <line x1="364" y1="128" x2="364" y2="224" stroke="#FFE680" stroke-width="3" stroke-linecap="round" opacity="0.8" />
    <path d="M 376 128 L 414 128 C 428 128 438 136 438 148" fill="none" stroke="#FFE680" stroke-width="2.5" stroke-linecap="round" opacity="0.8" />


    <!-- BOTTOM ROW: MAP -->

    <!-- LETTER M 3D SHADOW -->
    <polygon points="64,270 120,300 98,394 64,394" fill="#FF5E00" />
    <polygon points="120,300 150,270 186,310 186,394 154,394 126,350" fill="#FF5E00" />
    <!-- LETTER M FRONT FACE -->
    <polygon points="88,300 126,380 134,380 162,310 198,310 198,412 166,412 166,358 136,400 120,400 92,354 92,412 64,412 64,300" 
             fill="#FFB800" stroke="#FF7700" stroke-width="4" stroke-linejoin="miter" />
    <line x1="72" y1="310" x2="72" y2="404" stroke="#FFE680" stroke-width="3" stroke-linecap="round" opacity="0.8" />
    <line x1="88" y1="310" x2="124" y2="390" stroke="#FFE680" stroke-width="2.5" stroke-linecap="round" opacity="0.8" />
    <line x1="162" y1="316" x2="190" y2="316" stroke="#FFE680" stroke-width="2.5" stroke-linecap="round" opacity="0.8" />
    <line x1="188" y1="320" x2="188" y2="404" stroke="#FFE680" stroke-width="3" stroke-linecap="round" opacity="0.8" />

    <!-- LETTER A 3D SHADOW -->
    <polygon points="254,270 338,400 188,400" fill="#FF5E00" />
    <!-- LETTER A FRONT FACE -->
    <polygon points="274,278 348,412 218,412" fill="#FFB800" stroke="#FF7700" stroke-width="4" stroke-linejoin="miter" />
    <polygon points="274,324 316,396 242,396" fill="none" stroke="#FF5E00" stroke-width="3" stroke-linejoin="miter" />
    <circle cx="276" cy="366" r="8" fill="#FF5E00" />
    <circle cx="276" cy="366" r="4.5" fill="#FFE680" />
    <line x1="272" y1="290" x2="230" y2="404" stroke="#FFE680" stroke-width="3.5" stroke-linecap="round" opacity="0.8" />
    <line x1="230" y1="406" x2="336" y2="406" stroke="#FFE680" stroke-width="2" stroke-linecap="round" opacity="0.7" />

    <!-- LETTER P BOTTOM 3D SHADOW -->
    <path d="M 338 276 L 418 276 C 444 276 458 294 458 320 C 458 346 444 362 418 362 L 382 362 L 382 398 L 338 398 Z" 
          fill="#FF5E00" />
    <!-- LETTER P BOTTOM FRONT FACE -->
    <path d="M 356 300 L 416 300 C 438 300 452 314 452 334 C 452 354 438 368 416 368 L 392 368 L 392 412 L 356 412 Z" 
          fill="#FFB800" stroke="#FF7700" stroke-width="4" stroke-linejoin="miter" />
    <ellipse cx="410" cy="334" rx="18" ry="18" fill="#141724" stroke="#FF7700" stroke-width="3.5" />
    <circle cx="410" cy="334" r="7" fill="#FF5E00" />
    <circle cx="410" cy="334" r="4" fill="#FFE680" />
    <line x1="364" y1="308" x2="364" y2="404" stroke="#FFE680" stroke-width="3" stroke-linecap="round" opacity="0.8" />
    <path d="M 376 308 L 414 308 C 428 308 438 316 438 328" fill="none" stroke="#FFE680" stroke-width="2.5" stroke-linecap="round" opacity="0.8" />

  </g>
</svg>
`;

async function generateAllIcons() {
  console.log('Generating Hop-Map branding icons from SVG vector...');

  const targets = [
    { name: 'favicon-16x16.png', size: 16 },
    { name: 'favicon-32x32.png', size: 32 },
    { name: 'apple-touch-icon.png', size: 180 },
    { name: 'android-chrome-192x192.png', size: 192 },
    { name: 'icon-192.png', size: 192 },
    { name: 'android-chrome-512x512.png', size: 512 },
    { name: 'icon-512.png', size: 512 },
    { name: 'favicon.png', size: 512 }
  ];

  const svgBuffer = Buffer.from(svgLogo);

  // Ensure public directory exists
  if (!fs.existsSync('public')) {
    fs.mkdirSync('public', { recursive: true });
  }

  for (const t of targets) {
    const pngBuffer = await sharp(svgBuffer)
      .resize(t.size, t.size, { fit: 'contain' })
      .png({ compressionLevel: 9 })
      .toBuffer();

    // Write to root
    fs.writeFileSync(t.name, pngBuffer);
    // Write to public/
    fs.writeFileSync(path.join('public', t.name), pngBuffer);

    console.log(`✓ Generated ${t.name} (${t.size}x${t.size}) in root and public/`);
  }

  // Also create favicon.ico
  const icoBuffer = await sharp(svgBuffer)
    .resize(32, 32, { fit: 'contain' })
    .png()
    .toBuffer();
  fs.writeFileSync('favicon.ico', icoBuffer);
  fs.writeFileSync('public/favicon.ico', icoBuffer);
  console.log('✓ Generated favicon.ico in root and public/');

  console.log('All icons generated successfully!');
}

generateAllIcons().catch(err => {
  console.error('Error generating icons:', err);
  process.exit(1);
});
